src/rpdk/guard_rail/core/runner.py

```diff
@@ -183,12 +183,13 @@ def __execute__(schema_exec, ruleset):
             output = schema_exec(rules)
         return output
 
-    schema_to_execute = __exec_rules__(
-        schema=schema_diff(
-            previous_json=payload.previous_schema, current_json=payload.current_schema
-        )
+    schema_difference = schema_diff(
+        previous_json=payload.previous_schema, current_json=payload.current_schema
     )
+
+    schema_to_execute = __exec_rules__(schema=schema_difference)
     output = __execute__(schema_exec=schema_to_execute, ruleset=ruleset)
+    output.schema_difference = schema_difference
     compliance_output.append(output)
 
     return compliance_output
```

Comment:
Why does the output return List[GuardRuleSetResult] when we only one entry in the list ?
Shouldn't it reutrn GuardRuleSetResult ?